remote: Document HTTP remote push changes (#1006)
iterative/dvc.org: public/static/docs/command-reference/remote/modify.md

</details>

<details>

### Click for HTTP

- `basic_auth` - whether or not to use HTTP basic access authentication when
  accessing a remote.

  ```dvc
  $ dvc remote modify myremote basic_auth true
  ```

- `digest_auth` - whether or not to use HTTP digest access authentication when
  accessing a remote.

  ```dvc
  $ dvc remote modify myremote digest_auth true
  ```

  Note that `digest_auth` takes precedence over `basic_auth` if both options
  are enabled.

> - `user` - username to use to access a remote. The order in which dvc searches

Reviewer: dvc -> DVC (please, check other places)

  for username:

>   1. `user` specified in one of the dvc configs;

Reviewer: dvc -> DVC

  2. `user` specified in the url(e.g. `http://user@example.com/path`);

  ```dvc
  $ dvc remote modify myremote user myuser
  ```

> - `password` - a password to use to use when accessing a remote.

Reviewer: let's use `--local` here to avoid saving passwords into Git accidentally, check other remotes please

Author: fixed in 6ebcf5b

  ```dvc
  $ dvc remote modify myremote password mypassword
  ```

> - `ask_password` - ask for a password to use when accessing a remote.

Reviewer: can we clarify what happens if password is already provided?

Reviewer: I mean the password field above

Author: fixed in 6ebcf5b

</details>

<details>

### Click for HTTP

- `auth` - authentication method to use when accessing a remote. Accepted values
  are:

  - `basic` - use HTTP basic access authentication (`user` and `password`

>     options should also be provided).

Reviewer 2: ``[Basic authentication scheme](https://tools.ietf.org/html/rfc7617). `user` and `password` (or `ask_password`) parameters should also be configured.``

  - `digest` - use HTTP digest access authentication (`user` and `password`

>     options should also be provided).

Reviewer 2: ``[Digest Access Authentication Scheme](https://tools.ietf.org/html/rfc7616). `user` and `password` (or `ask_password`) parameters should also be configured.``

  - `custom` - use a custom HTTP header field and value when accessing a remote
    (`custom_auth_header` and `password` options should also be provided). When
    `auth` is set to `custom`, an additional HTTP header field in the form of
    `custom_auth_header: password` will be set for all HTTP requests to a

>     remote.

Reviewer 2: ``An additional HTTP header field will be set for all HTTP requests to the remote in the form: `custom_auth_header: password`. `custom_auth_header` and `password` (or `ask_password`) parameters should also be configured. ``

Not sure about "(or `ask_password`)" here.

Author: `ask_password` can technically still be used to set the password/header field value here, so I'll go ahead and document it

Reviewer 2: Thanks!

  ```dvc
  $ dvc remote modify myremote auth basic
  ```

- `custom_auth_header` - header field name to use to access a remote (only

>   applicable when `auth` is set to `custom`).

Reviewer 2: ``HTTP header field name to use when the `auth` parameter is set to `custom`.``

  ```dvc
  $ dvc remote modify myremote custom_auth_header My-Header
  ```

- `user` - username to use to access a remote. The order in which DVC searches

>   for username:

Reviewer 2: ``user name to use when the `auth` parameter is set to `basic`. The order...``

  1. `user` specified in one of the DVC configs;
  2. `user` specified in the url(e.g. `http://user@example.com/path`);

  ```dvc
  $ dvc remote modify myremote user myuser
  ```

- `password` - a password or custom header value to use to use when accessing a

>   remote.

Reviewer 2: ``password to use for any `auth` method.``

  ```dvc
  $ dvc remote modify myremote --local password mypassword
  ```

  > Note that the specified password will be inserted into the `.dvc/config`

>   > file. Therefore, it is recommended to add the password string with the

Reviewer 2: ```suggestion
  > file. Therefore, it's recommended to configure it using the
```

>   > `--local` option, enforcing it to be written to a Git-ignored config file

Reviewer 2: ```suggestion
  > `--local` option, which writes it to a Git-ignored config file.
```

>   > (or to use the `ask_password` option instead).

Reviewer 2: ```suggestion
  > (Or use the `ask_password` parameter instead.)
```

- `ask_password` - ask for a password or custom header value to use when

>   accessing a remote.

Reviewer 2: ``ask each time for the password to use for any `auth` method.``

  ```dvc
  $ dvc remote modify myremote ask_password true
  ```

  > Note that the `password` option takes precedence over `ask_password`. If
  > `password` is specified, DVC will not prompt the user to enter a password at

>   > runtime.

Reviewer 2: `... DVC will not prompt the user to enter a password for this remote.`
